android/apollo/src/com/andrew/apollo/provider/RecentSongStore.java

```diff
@@ -0,0 +1,161 @@
+package com.andrew.apollo.provider;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.text.TextUtils;
+
+/**
+ * Created by votaguz on 1/15/18.
+ */
+
+public final class RecentSongStore extends SQLiteOpenHelper {
+
+    /* Version constant to increment when the database should be rebuilt */
+    private static final int VERSION = 1;
+
+    /* Name of database file */
+    public static final String DATABASE_NAME = "songhistory.db";
+    private static RecentSongStore sInstance = null;
+    private final SQLiteDatabase writableDatabase;
+    private final SQLiteDatabase readableDatabase;
+
+    /**
+     * Constructor of <code>RecentStore</code>
+     *
+     * @param context The {@link Context} to use
+     */
+    public RecentSongStore(final Context context) {
+        super(context, DATABASE_NAME, null, VERSION);
+        writableDatabase = getWritableDatabase();
+        writableDatabase.enableWriteAheadLogging();
+        readableDatabase = getReadableDatabase();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void onCreate(final SQLiteDatabase db) {
+        db.execSQL(
+                "CREATE TABLE IF NOT EXISTS "
+                + RecentStoreColumns.TABLE_NAME + " ("
+                + RecentStoreColumns.ID + " LONG NOT NULL,"
+                + RecentStoreColumns.SONG_NAME + " TEXT NOT NULL,"
+                + RecentStoreColumns.ARTIST_NAME + " TEXT NOT NULL,"
+                + RecentStoreColumns.ALBUM_NAME + " TEXT,"
+                + RecentStoreColumns.DURATION + " LONG NOT NULL,"
+                + RecentStoreColumns.TIME_PLAYED + " LONG NOT NULL);"
+        );
+    }
+
+    @Override
+    public void onUpgrade(final SQLiteDatabase db, final int oldVersion, final int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS " + RecentStoreColumns.TABLE_NAME);
+        onCreate(db);
+    }
+
+    public static synchronized RecentSongStore getInstance(final Context context) {
+        if (sInstance == null) {
+            sInstance = new RecentSongStore(context.getApplicationContext());
+        }
+        return sInstance;
+    }
+
+    public void addSongId(final Long songId, final String songName,
+                          final String artistName, final String albumName,
+                          final long duration) {
```

Comment:
pass `,final String unknownString)` as the last parameter, to use in case any of `[songName, artistName, albumName]` are `null`